lambda-deployment-orca/src/main/java/com/amazon/aws/spinnaker/plugin/lambda/traffic/LambdaTrafficUpdateVerificationTask.java

```diff
@@ -64,7 +68,38 @@ public TaskResult execute(@NotNull StageExecution stage) {
             return formErrorTaskResult(stage,op.getErrors().getMessage());
         }
 
+        if (!"$WEIGHTED".equals(stage.getContext().get("deploymentStrategy"))) {
+            boolean valid = validateWeights(stage);
+            if (!valid) {
+                formErrorTaskResult(stage, "Could not update weights in time");
+                return TaskResult.builder(ExecutionStatus.TERMINAL).outputs(stage.getOutputs()).build();
+            }
+        }
+
         copyContextToOutput(stage);
         return taskComplete(stage);
     }
+
+    private boolean validateWeights(StageExecution stage) {
+        utils.await(30000);
```

Comment:
How'd you choose 30000 here?  Better to make it configurable?

Reply:
Is it possible/better to use stage timeout information instead of making up new timeout properties?